### DISCUSSION TITLE
Consulta firebase passando lista como parametros

### DISCUSSION THREAD

**Discussion by brownjrfla** [frontendbr/forum#1051] [Perguntas e Respostas]
Galera, 
Boa tarde!
Isto é usando para trazer todos os usuários

this.users = this.db.list(`/users`,
(ref: firebase.database.Reference) => ref.limitToLast(30).orderByChild('name')
);

Mas se eu quisesse fazer isso  para não trazer todos e sim apenas o que estão na lista
tipo isso

listausuarios 
this.users = this.db.list(`/users/${listausuarios} `,
(ref: firebase.database.Reference) => ref.limitToLast(30).orderByChild('name')
);

Vamos supor que no firebase tivesse os usuários:
AAAAAAA
BBBBBBB
CCCCCC
DDDDDD
ZZZZZZZ
E na minha lista eu só mandasse  
listausuarios  {BBBBBBB,  DDDDDD, ZZZZZZZ}
É possivel?

**Comment by klarkc** [comment]
Acho que você teria de fazer assim:

```js
const filteredUsers = []
const usersRef = this.db.child("users")
listusuarios.forEach(
    user => usersRef.orderByKey().equalTo(user).on(value => filteredUsers.push(value))
)
```

ou assim

```js
const filteredUsers = []
const usersRef = this.db.child("users")
listusuarios.forEach(
    user => usersRef.child(user).on(value => filteredUsers.push(value))
)
```

**Comment by brownjrfla** [comment]
Vou testar estas duas formas @klarkc e passo um feedback

**Comment by brownjrfla** [comment]
Fiz um foreach mesmo mas não consegui colocar no observable que criei
Users: Observable<User[]>:
Ele sempre ficava com o ultimo valor na hora de listar 
Então criei um let User[];
E usei no foreach
E após isso atribui o valor do let User[]  ao Users: Observable<User[]>
E funcionou a lista com todos os itens
Não me parece elegante 
mas por hora me ajudou
